Kinvey.Core/Core/NetworkFactory.cs

```diff
@@ -286,7 +286,8 @@ public NetworkRequest<T> BuildDeltaSetRequest<T>(string collectionName, string l
             {
                 restPathParams += string.IsNullOrEmpty(restPathParams) ? "?" : "&";
                 restPathParams += "query={query}";
-                urlParameters.Add("query", query);
+                var encodedQuery = System.Net.WebUtility.UrlEncode(query);
```

Comment:
@romanogolikhin I'm not entirely sure if we should use `System.Net.WebUtility.UrlEncode()` or `Uri.EscapeDataString()`. Reading the docs and some links in the internet it seems very confusing the difference between those 2. I'm wondering if you have an opinion about it.

Reply:
not 100% sure yet, but it seems `System.Uri.EscapeDataString()` is what we need in this case

https://stackoverflow.com/questions/602642/server-urlencode-vs-httputility-urlencode

Reply:
@heyzooi  I have read what you had recommended me. I agree with you.
The "System.Uri.EscapeDataString()" is more preferable for us.
I have committed changes.